[RA-1 Ch07] LCM Content (#1182)
cntt-n/CNTT: doc/ref_arch/openstack/chapters/chapter07.md

<p align="right"><img src="../figures/bogo_ifo.png" alt="scope" title="Scope" width="35%"/></p>

## Table of Contents
* [7.1 Introduction ](#7.1)
* [7.2 Requirements ](#7.2)
* [7.3 NFVI and VIM configuration management ](#7.3)

> * [7.4 MFVI and VIM Maintenance ](#7.4)

Reviewer: @pgoyal01 i think NFVI not MFVI

Author: @ASawwaf Thanks -- Corrected

- Provisioning of host aggregates

<a name="7.4"></a>

> ## 7.4 MFVI and VIM Maintenance

Reviewer: @pgoyal01 i think NFVI not MFVI

Author: @ASawwaf Thanks - Corrected

- Backup, upgrade and rollback
-	Capacity management, horizontal scaling by adding bare metal servers(control/compute/storage)
-	Solution evolution by adding new OpenStack services or additional components

<a name="7.5"></a>
## 7.5 Logging, Monitoring and Analytics
- Logging
- Monitoring 
- Alerting
- Logging, Monitoring, and Analytics (LMA) Framework

> ### 7.5.1. Logging

Reviewer 2: Please add these subheadings in ToC

Author: @deebhatia Done - Thanks

of events relevant to a particular system. All software, including operating systems, middleware and 
applications produce log files. 
Enterprises and vendors may have custom monitoring and logging solutions. 
The intent of the logging and monitoring is to capture events and 
data of interest to the NFVI and workloads so that appropriate actions can be taken. For example, 

- Operating systems and web servers maintain an access log of all access requests, session details and file access.
- Databases maintain a transaction log of all transaction executed including an added, changed and deleted data.
- Audit logs record chronological documentation of any activities that could have affected a 
particular operation or event. Data typically includes resources accessed, destination and source 
addresses, and a timestamp and login information for the person who accessed the resources.

Some of the data is to support the metrics collection specified in the [Reference Model Chapter 4: Infrastructure Capabilities, Metrics and Catalogue](../../ref_model/chapters/chapter04.md).

Logs have multiple operational uses including for:

> 1. Regulatory Compliance and SIEM featuring the automated gathering, analysis and correlation of log data across all systems and devices across AT&T to provide real-time analysis, event prioritization, reporting, notification and alerting.

Reviewer 3: ```suggestion
1. Regulatory Compliance and SIEM featuring the automated gathering, analysis and correlation of log data across all systems and devices across an operator to provide real-time analysis, event prioritization, reporting, notification and alerting.
```

Author: @kedmison Corrected -- Thanks

### 7.5.2. Monitoring
Monitoring is the process of collecting, aggregating, and analyzing values that improve awareness of 
the components' characteristics and behavior. The data from various parts of the environment are collected 
into a monitoring system that is responsible for storage, aggregation, visualization, and initiating automated 
responses when the values meet specific threshold.

Monitoring systems fulfill many related functions. Their first responsibility is to accept and store incoming 
and historical data. While values representing the current point in time are useful, it is almost always more 
helpful to view those numbers in relation to past values to provide context around changes and trends. 

### 7.5.2. Alerting
Alerting is the responsive component of a monitoring system that performs actions based on changes in metric 
values. Alert definitions are composed of two components: a metrics-based condition or threshold, and an 
action to perform when the values fall outside of the acceptable conditions.

While monitoring systems are incredibly useful for active interpretation and investigation, one of the primary

> benefits of a complete monitoring system is letting administrators disengage from the system. ALerts allow the

Reviewer 3: ```suggestion
benefits of a complete monitoring system is letting administrators disengage from the system. Alerts allow the
```

Author: @kedmison Corrected -- Thanks

specification of situations that make sense to actively manage, while relying on the passive monitoring of the 
software to watch for changing conditions.

### 7.5.3. Logging,Monitoring, and Analytics (LMA) Framework

> In this section, a possible framework utilizing Prometheus, Elasticsearch and Kibana is given as an example only.

Reviewer 3: Can and should this note be extended to cover fluentd and prometheus (referred to below)?

Author: @kedmison Corrected - Thanks

> <p align="center"><img src="../figures/Figure_4_4_Monitoring_Logging_Framework.png" alt="Monitoring and Logging Framework"></br>

Reviewer 4: The diagram uses an undefined abbreviation 'PVC'

Reviewer 3: The diagram also refers to Pods, suggesting a particular technology stack is in use.

Author: @iangardner22 @kedmison Deleted original Figure $-4 and replaced with updated Figure 7-1. Thanks

> <p align="center"><img src="../figures/Figure_7_1_Monitoring_Logging_Framework.png" alt="Monitoring and Logging Framework"><b>

Reviewer 2: The image has a typo for word `visualisation`

Author: @deebhatia Corrected - Thanks

Figure 7-1: Monitoring and Logging Framework </p>

The monitoring and logging framework (**Figure 7-1**) leverages Prometheus as the monitoring engine and 
Fluentd for logging. In addition, the framework uses Elasticsearch to store and organize logs for easy access. 
Prometheus agents pull information from individual components on every host.  Fluentd, an open source data 
collector, unifies data collection and consumption for better use and understanding of data. Fluentd captures 

> the access, application and system logs.

Reviewer 4: Where do we reference PM - Performance Management?   Do we have a suggested model with collectd and some central collection?

Author: @iangardner22 Atthis time we only have requirements but we do need a model as you have suggested.
